### PR TITLE
ENH: lgr.debug that we are using DANDI_API_KEY provided key

### PR DESCRIPTION
**Pull request by yarikoptic**
to possibly avoid confusions like
https://github.com/dandi/dandi-cli/issues/1130

or should we go for `.info`?

### CODE REVIEW

File: dandi/dandiapi.py [hunk 1]
```diff
@@ -474,6 +474,7 @@ def dandi_authenticate(self) -> None:
         # Shortcut for advanced folks
         api_key = os.environ.get("DANDI_API_KEY", None)
         if api_key:
+            lgr.debug("Using api key from DANDI_API_KEY env variable")
```
Comment (jwodder) [inline dandi/dandiapi.py:477]:
```suggestion
            lgr.debug("Using API key from DANDI_API_KEY environment variable")
```

**Review by jwodder — CHANGES_REQUESTED** [review]
You might want to add corresponding "Using API key from keyring/user input" log messages around line 492.

File: dandi/dandiapi.py (hunk 1 above)
Comment (yarikoptic) [reply dandi/dandiapi.py:477]: will do, thanks

**Comment by yarikoptic** [comment]
> You might want to add corresponding "Using API key from keyring/user input" log messages around line 492.

good idea, did that+ -- please check

File: dandi/dandiapi.py (hunk 1 above)
Comment (jwodder) [reply dandi/dandiapi.py]: @yarikoptic You didn't commit my suggested change.

**Review by jwodder — APPROVED** [review]

**Pull request merged**